src/compas_eve/mqtt/mqtt_paho.py

```diff
@@ -60,8 +60,8 @@ def publish(self, topic, message):
         """
 
         def _callback(**kwargs):
-            json_message = topic._message_to_json(message)
-            self.client.publish(topic.name, json_message)
+            encoded_message = self.codec.encode(message)
+            self.client.publish(topic.name, encoded_message)
```

Comment:
Based on the comments above, this would change to call the conversion to data from the topic, and then use the codec for encoding:

```suggestion
            data = topic._message_to_data(message)
            encoded_message = self.codec.encode(data)
            self.client.publish(topic.name, encoded_message)
```

Reply:
the same would need to be done in `mqtt_cli.py` for this to work on IronPython